source/common/router/route_config_update_impl.cc

```diff
@@ -0,0 +1,98 @@
+#include "common/router/route_config_update_impl.h"
+
+#include <string>
+
+#include "envoy/api/v2/route/route.pb.validate.h"
+
+#include "common/common/assert.h"
+#include "common/common/fmt.h"
+#include "common/protobuf/utility.h"
+#include "common/router/config_impl.h"
+
+namespace Envoy {
+namespace Router {
+
+bool RouteConfigUpdateImpl::onRdsUpdate(const envoy::api::v2::RouteConfiguration& rc,
+                                        const std::string& version_info) {
+  const auto start_of_update = time_source_.systemTime();
+
+  const uint64_t new_hash = MessageUtil::hash(rc);
+  if (new_hash == last_config_hash_) {
+    return false;
+  }
+
+  route_config_proto_ = rc;
+  last_config_hash_ = new_hash;
+  last_config_version_ = version_info;
+  last_updated_ = start_of_update;
```

Comment:
Could just inline the `start_of_update` definition here, this doesn't need to be precise.